src/argprocessing.hpp

```diff
@@ -18,22 +18,46 @@
 
 #pragma once
 
+#include "Args.hpp"
 #include "stats.hpp"
 
 #include "third_party/nonstd/optional.hpp"
 
 class Context;
 class Args;
 
+struct ProcessArgsResult
+{
+  ProcessArgsResult(enum stats err) : error(err)
+  {
+  }
+  ProcessArgsResult(Args preprocessor_args,
+                    Args extra_args_to_hash,
+                    Args compiler_args)
+    : preprocessor_args(preprocessor_args),
+      extra_args_to_hash(extra_args_to_hash),
+      compiler_args(compiler_args)
+  {
+  }
+
+  // nullopt on success, otherwise the statistics counter that should be
+  // incremented.
+  nonstd::optional<enum stats> error;
+
+  // Arguments (except -E) to send to the preprocessor.
+  Args preprocessor_args;
+
+  // Arguments not sent to the preprocessor but that should be part of the
+  // hash.
+  Args extra_args_to_hash;
+
+  // Arguments to send to the real compiler.
+  Args compiler_args;
+};
+
 // Process the compiler options into options suitable for passing to the
 // preprocessor (`preprocessor_args`) and the real compiler (`compiler_args`).
 // `preprocessor_args` doesn't include -E; this is added later.
 // `extra_args_to_hash` are the arguments that are not included in
 // `preprocessor_args` but that should be included in the hash.
-//
-// Returns nullopt on success, otherwise the statistics counter that should be
-// incremented.
-nonstd::optional<enum stats> process_args(Context& ctx,
-                                          Args& preprocessor_args,
-                                          Args& extra_args_to_hash,
-                                          Args& compiler_args);
+ProcessArgsResult process_args(Context& ctx);
```

Comment:
Actually they still make sense. The variables are named exactly like that within the result struct.

Reply:
The comment style we have used lately describes the function parameters within backticks. Since there are no such parameters, the comment is incorrect in a formal sense. But regardless, the information in the comment is now redundant since the variables are better described in `struct ProcessArgsResult`, so I suggest applying DRY and just remove them from the comment.

Reply:
That's reasonable. It's really kind of the same text.